### ISSUE TITLE
Store language preference in the user's profile instead of cookies

### ISSUE THREAD

**Issue by arnoldnipper** [peeringdb/peeringdb#1247]
**Is your feature request related to a problem? Please describe.**
Every time I've to set the language preferences again. Would be nice to store the preference in the user's profile.

**Comment by grizz** [comment]
+1

**Comment by mcmanuss8** [comment]
+1

**Comment by netravnen** [comment]
+1

**Comment by dvanallen** [comment]
I looked at this a bit while working on #1295.

I confirmed that we do already save the user's language selection from the profile view to the database in the user table's `locale` field.

We also read that value back out, but only while processing the login view.  At that time we set a session-length language cookie.  Setting the language in the profile also creates a new session-length language cookie.

The session-length language cookies will generally expire when the user closes their browser.  However, the user will still have a session cookie, and associated authenticated session on the server.  So when they revisit the site, they will not need to visit the login page, and they will not have a new language cookie generated.  I believe this explains the behavior you described.

The change in #1295 will help a bit here, as the new select box will also create a new language cookie for the user, which makes it a bit easier to work around this issue.

One other improvement here could be to add some additional time to the expiration of the language cookies, to make them last longer than the session cookies.
